src/main/kotlin/lotto/domain/LottoList.kt

```diff
@@ -0,0 +1,26 @@
+package lotto.domain
+
+import lotto.domain.lottoStrategy.LottoStrategy
+
+data class LottoList(
+    val lottos: List<Lotto>
+) {
+    fun size(): Int = lottos.size
+
+    fun getResult(previousLottoNumbers: LottoNumbers): LottoResult =
+        LottoResult(
+            lottos
+                .groupBy { it.countMatchNumbers(previousLottoNumbers) }
+                .mapValues { LottoList(it.value) }
+        )
+
+    companion object {
+        fun of(lottoStrategy: LottoStrategy, lottoCount: Long): LottoList {
+            return LottoList(
+                (1..lottoCount).map {
+                    Lotto(lottoStrategy.makeLottoNumbers())
+                }
```

Comment:
List fake constructor를 활용해볼 수 있겠네요 :)
```suggestion
                List(lottoCount) {
                    Lotto(lottoStrategy.makeLottoNumbers())
                }
```